lib/Sema/CSApply.cpp

```diff
@@ -6417,7 +6461,8 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
   }
 
   // Unresolved types come up in diagnostics for lvalue and inout types.
-  if (fromType->hasUnresolvedType() || toType->hasUnresolvedType())
+  if (fromType->hasUnresolvedType() || toType->hasUnresolvedType() ||
+      fromType->is<ErrorType>())
```

Comment:
Is there a test case that triggers this? You should split it out into a separate patch with a commit message explaining why

Reply:
Removed, thanks.